popvae.py

```diff
@@ -0,0 +1,589 @@
+                          ` --parallel --gpu_number 0,1,2,3` to split batches across 4 GPUs.\
+                          default: 0',
+)
+parser.add_argument(
+    "--prediction_freq",
+    default=5,
+    type=int,
+    help="print predictions during training every \
+                          --prediction_freq epochs. default: 10",
+)
+parser.add_argument(
+    "--max_SNPs",
+    default=None,
+    type=int,
+    help="If not None, randomly select --max_SNPs variants \
+                          to run. default: None",
+)
+parser.add_argument(
+    "--latent_dim", default=2, type=int, help="N latent dimensions to fit. default: 2"
+)
+parser.add_argument(
+    "--PCA",
+    default=False,
+    action="store_true",
+    help="Run PCA on the derived allele count matrix in scikit-allel.",
+)
+parser.add_argument(
+    "--n_pc_axes",
+    default=20,
+    type=int,
+    help="Number of PC axes to save in output. default: 20",
+)
+parser.add_argument(
+    "--PCA_scaler",
+    default="Patterson",
+    type=str,
+    help="How should allele counts be scaled prior to running the PCA?. \
+                          Options: 'None' (mean-center the data but do not scale sites), \
+                          'Patterson' (mean-center then apply the scaling described in Eq 3 of Patterson et al. 2006, Plos Gen)\
+                          default: Patterson. See documentation of allel.pca for further information.",
+)
+parser.add_argument(
+    "--plot",
+    default=False,
+    action="store_true",
+    help="generate an interactive scatterplot of the latent space. requires --metadata. Run python scripts/plotvae.py --h for customizations",
+)
+parser.add_argument(
+    "--metadata",
+    default=None,
+    help="path to tab-delimited metadata file with column 'sampleID'.",
+)
+args = parser.parse_args()
+
+infile = args.infile
+save_allele_counts = args.save_allele_counts
+patience = args.patience
+batch_size = args.batch_size
+max_epochs = args.max_epochs
+seed = args.seed
+save_weights = args.save_weights
+train_prop = args.train_prop
+gpu_number = args.gpu_number
+out = args.out
+prediction_freq = args.prediction_freq
+max_SNPs = args.max_SNPs
+latent_dim = args.latent_dim
+PCA = args.PCA
+PCA_scaler = args.PCA_scaler
+depth = args.depth
+width = args.width
+n_pc_axes = args.n_pc_axes
+search_network_sizes = args.search_network_sizes
+plot = args.plot
+metadata = args.metadata
+
+depth_range = args.depth_range
+depth_range = np.array([int(x) for x in re.split(",", depth_range)])
+width_range = args.width_range
+width_range = np.array([int(x) for x in re.split(",", width_range)])
+
+if args.plot:
+    if args.metadata == None:
+        print("ERROR: `--plot` argument requires `--metadata`")
+        exit()
+
+os.environ["CUDA_VISIBLE_DEVICES"] = gpu_number
+
+if not seed == None:
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    tensorflow.set_random_seed(seed)
```

Comment:
this is throwing an error on the example in the readme. I think this call has been changed to `tf.random.set_seed()`